### PR TITLE
WINC-607: [e2e] Skip MachineSet update in deletion test

### PR DESCRIPTION
**Pull request by jrvaldes**
This PR skips the MachineSet update in the Windows node deletion
test if no MachineSet is found with the Windows label. This is a
corner case for platform-agnostic infrastructure and e2e tests
executions where no instances are configured via MachineController, for
example: `hack/run-ci-e2e-test.sh -m 0`.

### CODE REVIEW

File: test/e2e/delete_test.go [hunk 1]
```diff
@@ -115,7 +115,10 @@ func testWindowsNodeDeletion(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// set excepted node count to zero, since all Windows Nodes should be deleted in the test
+	expectedNodeCount := int32(0)
```
Comment (saifshaikh48) [inline test/e2e/delete_test.go:119]: nit: I'd prefer this initialized closer to its usage (around line 132)

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

minor comment which shouldn't prevent merging IMO

File: test/e2e/delete_test.go (hunk 1 above)
Comment (jrvaldes) [reply test/e2e/delete_test.go:119]: True and agree. I moved it up here to facilitate a more natural flow to the `MachineSets` logic. 
- [Fetch](https://github.com/openshift/windows-machine-config-operator/blob/1b7feee90c711ce964ef3be75ddf3baf2f6a12e7/test/e2e/delete_test.go#L122) the MachineSets
- [Filter](https://github.com/openshift/windows-machine-config-operator/blob/1b7feee90c711ce964ef3be75ddf3baf2f6a12e7/test/e2e/delete_test.go#L126) the MachineSets
- [Check](https://github.com/openshift/windows-machine-config-operator/blob/1b7feee90c711ce964ef3be75ddf3baf2f6a12e7/test/e2e/delete_test.go#L133) the MachineSet with the Windows Label

Happy to move it closer, in case we need to.

File: test/e2e/delete_test.go [hunk 2]
```diff
@@ -126,20 +129,18 @@ func testWindowsNodeDeletion(t *testing.T) {
...
-		windowsMachineSetWithLabel, meta.UpdateOptions{})
-	require.NoError(t, err, "error updating Windows MachineSet")
-
-	// we are waiting 10 minutes for all windows machines to get deleted.
-	err = testCtx.waitForWindowsNodes(expectedNodeCount, true, false, false)
-	require.NoError(t, err, "Windows node deletion failed")
-
+	// skip the scale down step if there is no MachineSet with Windows label
+	if windowsMachineSetWithLabel != nil {
+		// Scale the Windows MachineSet to 0
```
Comment (selansen) [inline test/e2e/delete_test.go:134]: This comment was more meaningful where we used to set `expectedNodeCount := int32(0)` next line. It would be good if you can move it back here.

**Comment by sebsoto** [comment]
@jrvaldes please run the platform none test on this PR to ensure the correct behavior is being seen

**Comment by jrvaldes** [comment]
> @jrvaldes please run the platform none test on this PR to ensure the correct behavior is being seen

For the test to pass needs https://github.com/openshift/windows-machine-config-operator/pull/900 and https://github.com/openshift/windows-machine-config-operator/pull/900

**Comment by jrvaldes** [comment]
/close 

Closing this PR to cover all e2e test adjustments for `platform-none` in https://github.com/openshift/windows-machine-config-operator/pull/896, where the `platform-none-vsphere-e2e-operator` job must pass.
